presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -673,9 +681,10 @@ protected String toRemoteTableName(JdbcIdentity identity, Connection connection,
     {
         requireNonNull(remoteSchema, "remoteSchema is null");
         requireNonNull(tableName, "tableName is null");
-        verify(CharMatcher.forPredicate(Character::isUpperCase).matchesNoneOf(tableName), "Expected table name from internal metadata to be lowercase: %s", tableName);
 
         if (caseInsensitiveNameMatching) {
+            log.warn("The 'case-insensitive-name-matching' config is deprecated and may lead to query issues. " +
+                    "Please use 'case-sensitive-name-matching=true' for proper case handling.");
```

Comment:
At what frequency is this logged? Every query? I wonder if we could instead add this to the session's warning collector.

Reply:
I think we can also more clearly steer users away from this config by more clearly warning them about the consequences of using this configuration

Consider this message instead:

> 'case-insensitive-name-matching' is deprecated. Use of this configuration value may lead to inconsistent behavior for schema, table, and column name matching. Please switch to using 'case-sensitive-name-matching' for proper case sensitivity behavior.

Also, in this block if `case-insensitive-name-matching` is true, wouldn't 'case-sensitive-name-matching=true` as the recommendation be the opposite of what the user is trying to configure?